apps/prairielearn/src/middlewares/selectAndAuthzAssessment.html.ts

```diff
@@ -0,0 +1,33 @@
+import { html } from '@prairielearn/html';
+
+import { HeadContents } from '../components/HeadContents.html.js';
+import { Navbar } from '../components/Navbar.html.js';
+import { config } from '../lib/config.js';
+
+export function AccessDenied({ resLocals }: { resLocals: Record<string, any> }) {
+  const courseInstance = resLocals.course_instance.id;
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${HeadContents({ resLocals })}
+      </head>
+      <body>
+        ${Navbar({ resLocals, navPage: 'assessment_instance' })}
+        <main id="content" class="container">
+          <h1>Access Denied</h1>
```

Comment:
I'm wondering whether we should remove this header entirely? It still has a slightly error-like feel to it. I think the "not available" text below is clear, without looking scary.

I'd appreciate other's input here, especially @mfsilva22.

Reply:
agree that jus saying "this assessment is not available..." is sufficient. I would support removing the heading "Access denied"

Reply:
It looks like it's missing something without the header?

![image](https://github.com/user-attachments/assets/af923165-d156-452f-9ac5-21fd3eb0df1d)

Reply:
Maybe replace denied with something else, such as inaccessible?

Reply:
I agree that it looks better visually with the header. Saying "Assessment inaccessible" in a header does feel better than "Access denied", but it still seems to imply to me that there is an error. Like "we tried to access the assessment but it failed". I really like Dave's text ("This assessment is not available to you right now") because it sounds to me like this is intended behavior, which is what we are trying to communicate.

For a heading, ChatGPT suggested:
> Assessment Unavailable

ChatGPT also suggested a slight tweak of the message:
> This assessment is not currently available to you, based on the course settings.

I like the tweaked message above because it makes it even clearer that the lack of access is the intended behavior. I continue to lean towards not using a heading at all, because all of these header suggestions could easily be misinterpreted as indicating that there was an error.

Does anyone have other suggestions for the wording? We are trying to convey:
* Not accessible
* To you
* At the moment
* This is the intended behavior and is not an error

Reply:
I honestly like the original message, and I think "Assessment Unavailable" is a decent header. I really, really think we have to give users some credit and assume that they can reasonably interpret "not available right now" as not an error state. That said, I think "Access denied" _could_ be interpreted as an error, which is why I lean towards "Assessment unavailable".

I specifically don't like "based on the course settings" because if a student forwards that message to an instructor they might reasonable go to the course "Settings" tab to try to resolve it. We could of course say "based on the assessment's access rules", but I'd rather not leak that implementation detail to students. Spitballing another option: "This assessment's configuration does not allow you to access it right now"?

Reply:
The nice part of this being a page and not error handling is we could make the description as descriptive as we need it to be. How about something like:

![image](https://github.com/user-attachments/assets/5d3d60f7-88e3-4c43-8456-9e648eb64589)

We could continue to say "If you believe this is in error, contact your..." but I think that's default response anyway so we don't need to tell students to tell us. :D

Reply:
I think any of the text options we've had are fine and I'm not going to worry which exact wording we have in the text because none of them sound like errors.

I still think that the header text makes this whole page look like an error page. Seeing "Assessment Unavailable" (while better than "Access Denied") in large bold letters basically has the look and feel of an error page to me. If I tried to go to some random URL and got a message like "Page Unavailable" I think my first impulse would be that something went wrong with the web server. I don't think that I would think "oh yes, this is all fine".

To be clear, I'm quite happy to give users credit for understanding "not available to you right now". The thing I'm against is showing the big bold message in the header, because of the "look and feel" of the page. My model of most users is that they don't carefully read messages and think about them, they just get an immediate impression of "this matches my impression of an error page" and then they bounce.

All that being said, I'd like to hear @mfsilva22's opinion on this. I don't really trust the opinion of any of the other people on this thread (myself very much included) about what a regular user will think when seeing this page 😅

Reply:
What about "This assessment is closed" for the heading?

Overall, I am happy with the current version. Basically anything that is not black and red is an improvement :)

Reply:
"Closed" unfortunately already has a very specific meaning for assessments, and we already have another page that specifically handles that page. I'd rather avoid overloading terms so that we don't further confuse anyone.

On that note: the "closed" page uses a card with a blue card header. What if we do the same here? I think blue is a nice "everything is fine!" color, and blue cards specifically are in pretty much all of our "this is working as intended" pages.

Reply:
I think that is fine. As I said, from my point of view, I don't think the current page is scary (and we may find out more information once users start using it)

Reply:
I like the idea of a card with a blue header. This definitely looks like one of our regular pages and less like an error page. @nwalters512 I guess you are suggesting to put "Assessment unavailable" as the white card header and the text and buttons in the card?

What do you think, @trombonekenny? Maybe this is worth a try?

Reply:
Although given that @mfsilva22 has signed off on the current page I'm also happy to go with this.

Reply:
I lean towards using a blue header for consistency. For reference, the "closed" page I'm referring to is https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/src/middlewares/studentAssessmentAccess.html.ts.

Reply:
Yep, I like the card + blue header idea

Reply:
Latest draft:

![image](https://github.com/user-attachments/assets/9ee488a3-4909-4a22-afb1-42f4d0cb5bf4)

Definitely doesn't look like an error and is brief enough that I think people will get the idea?